### ISSUE TITLE
ch-run: `error: can't canonicalize: /var/tmp: Invalid argument (ch_misc.c:596 22)`

### ISSUE THREAD

**Issue by paulmenzel** [hpc/charliecloud#1644]
I am trying to run [parliament2](https://github.com/fritzsedlazeck/parliament2):

```
$ ch-image --version
0.31
$ ch-image pull dnanexus/parliament2:latest
[…]
$ ch-image list
dnanexus/parliament2:latest
$ ch-convert dnanexus/parliament2:latest parliament2.sqfs
input:   ch-image  dnanexus/parliament2:latest
output:  squash    parliament2.sqfs
packing ...
Parallel mksquashfs: Using 128 processors
Creating 4.0 filesystem on parliament2.sqfs, block size 65536.
[=================================================================================\] 191135/191135 100%

Exportable Squashfs 4.0 filesystem, gzip compressed, data block size 65536
	compressed data, compressed metadata, compressed fragments,
	compressed xattrs, compressed ids
	duplicates are removed
Filesystem size 1975660.19 Kbytes (1929.36 Mbytes)
	30.20% of uncompressed filesystem size (6541835.21 Kbytes)
Inode table size 1471946 bytes (1437.45 Kbytes)
	29.88% of uncompressed inode table size (4926539 bytes)
Directory table size 1357614 bytes (1325.79 Kbytes)
	43.25% of uncompressed directory table size (3139332 bytes)
Number of duplicate files found 33561
Number of inodes 125190
Number of files 101819
Number of fragments 8278
Number of symbolic links  12750
Number of device nodes 0
Number of fifo nodes 0
Number of socket nodes 0
Number of directories 10621
Number of ids (unique uids + gids) 1
Number of uids 1
	root (0)
Number of gids 1
	root (0)
4+0 records in
4+0 records out
4 bytes copied, 7.589e-05 s, 52.7 kB/s
done
$ ls -lh parliament2.sqfs
-rw-r----- 1 user user 1.9G Apr 27 17:03 parliament2.sqfs
$ ch-run parliament2.sqfs -- date
ch-run[23739]: error: can't canonicalize: /var/tmp: Invalid argument (ch_misc.c:596 22)
```

Without `ch-convert` I am getting the error below:

```
$ ch-image list
dnanexus/parliament2:latest
$ ch-run dnanexus/parliament2 -- echo hello
ch-run[24690]: error: can't stat: dnanexus/parliament2: No such file or directory (ch_core.c:255 2)
$ ch-run dnanexus/parliament2:latest -- echo hello
ch-run[24907]: error: can't canonicalize: //var/tmp/user.ch/img/dnanexus%parliament2+latest: Invalid argument (ch_misc.c:596 22)
```

**Comment by reidpr** [comment]
Hello, indeed this is a real bug, but I believe it's a duplicate of PR #1550 and thus should be fixed in 0.32.

Please do re-open if that's not it.

**Comment by paulmenzel** [comment]
Indeed, thank you. Charliecloud 0.32 works.
